loadtest/evm.go

```diff
@@ -0,0 +1,125 @@
+package main
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"encoding/hex"
+	"fmt"
+	"math"
+	"math/big"
+	"math/rand"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/ethclient"
+)
+
+type EvmTxSender struct {
+	nonceMap sync.Map
+	clients  []*ethclient.Client
+}
+
+func NewEvmTxSender(clients []*ethclient.Client) *EvmTxSender {
+	return &EvmTxSender{
+		nonceMap: sync.Map{},
+		clients:  clients,
+	}
+}
+
+// PrefillNonce is a function to fill starting nonce, this needs to be called at the beginning
+func (txSender *EvmTxSender) PrefillNonce(keys []cryptotypes.PrivKey) {
+	client := txSender.GetNextClient()
+	for _, key := range keys {
+		privKeyHex := hex.EncodeToString(key.Bytes())
+		privateKey, err := crypto.HexToECDSA(privKeyHex)
+		if err != nil {
+			fmt.Printf("Failed to load private key: %v \n", err)
+		}
+
+		publicKey := privateKey.Public()
+		publicKeyECDSA, ok := publicKey.(*ecdsa.PublicKey)
+		if !ok {
+			panic("Cannot assert type: publicKey is not of type *ecdsa.PublicKey \n")
```

Comment:
nit - does it make sense to panic here? or just print as warning?

Reply:
Yeah here we can just print a warning message